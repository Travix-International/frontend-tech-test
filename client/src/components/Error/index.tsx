import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faFrown } from '@fortawesome/free-solid-svg-icons';
import StateScreen from '@components/StateScreen';

const Icon = <FontAwesomeIcon icon={faFrown} size="4x" />;
const Error = () =>
    <StateScreen
        icon={Icon}
        text="Oops! Something is wrong."
    />


export default Error;