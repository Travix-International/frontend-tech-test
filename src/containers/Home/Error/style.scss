.home--error {
  opacity: 0;
  display: none;
  width: 95%;
  margin: 0 auto;

  .item--error {
    position: relative;
  }

  .ui-messageBox_error {
    .ui-messageBox__content {
      width: 100%;
      word-wrap: break-word;
    }
  }

  &.is-shown {
    opacity: 1;
    display: block;
  }

  .button--close {
    top: 0;
    background: none;
    border: none;
    -webkit-box-shadow: none;
    box-shadow: none;
    color: white;
    cursor: pointer;
    font-family: var(--tx-generic-font-primary-generic-family);
    font-size: var(--tx-sliding-panel-header-close-button-font-size);
    font-weight: var(--tx-generic-font-primary-weight-bold);
    height: var(--tx-sliding-panel-header-close-button-height);
    line-height: var(--tx-sliding-panel-header-close-button-height);
    outline: 0;
    overflow: visible;
    padding: 0;
    position: absolute;
    right: var(--tx-sliding-panel-header-close-button-right);
    text-align: right;
    width: var(--tx-sliding-panel-header-close-button-width);
  }
}
