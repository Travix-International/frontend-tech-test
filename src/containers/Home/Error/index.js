import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { get as _get } from 'lodash';
import classNames from 'classnames';
import {
  MessageBox,
} from 'travix-ui-kit';
import logger from '../../../logger';
import action from '../../../actions';

import './style.scss';

const HANDLING_ERRORS = [
  'POST_TASK',
  'PATCH_TASK',
  'DELETE_TASK',
];

export class MyComponent extends Component {
  static propTypes = {
    errorHandled: PropTypes.func.isRequired,
    errs: PropTypes.array.isRequired,
  }

  constructor(props) {
    super(props);

    this.state = {
      isShown: false,
    };
    this.onClick = this.onClick.bind(this);
  }

  onClick(e) {
    const errId = e.target.getAttribute('data-err-id');
    const { errorHandled } = this.props;
    errorHandled(errId);
    this.setState({
      isShown: false,
    });
  }

  componentWillReceiveProps(nextProps) {
    const { errs } = this.props;
    logger.debug('error:index', errs);

    // show err if there is
    if (errs.length !== nextProps.errs.length) {
      if (nextProps.errs.length > 0) {
        this.setState({
          isShown: true,
        });
      } else {
        this.setState({
          isShown: false,
        });
      }
    }
  }

  render() {
    const { isShown } = this.state;
    const { errs } = this.props;

    return (
      <div
        className={classNames('home--error', {
          'is-shown': isShown,
        })}
      >
        {errs.map((err) => {
          return (
            <div className="item--error">
              <MessageBox type="error">
                <p>Something went wrong</p>
                <p>{JSON.stringify(err)}</p>
              </MessageBox>
              <div className="button--close" data-err-id={err.id} onClick={this.onClick}>x</div>
            </div>
          );
        })}
      </div>
    );
  }
}

export function mapStateToProps(state) {
  const errors = _get(state, 'global.errors', []);
  const errs = [];
  errors.map((err) => {
    if (err.fromAction && HANDLING_ERRORS.indexOf(err.fromAction) > -1) {
      return errs.push(err);
    }
    return false;
  });

  return {
    errs,
  };
}

export function mapDispatchToProps(dispatch) {
  return {
    errorHandled: (errId) => {
      return dispatch(action.errorHandled(errId));
    },
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(MyComponent);
