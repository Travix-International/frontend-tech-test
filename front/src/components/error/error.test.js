jest.mock('react-dom')

import React from 'react'
import render from 'react-test-renderer'
import ShallowRenderer from 'react-test-renderer/shallow'

import Error from '.'

describe('ErrorComponent', () => {
  it('Snapshot', async () => {
    const renderer = new ShallowRenderer()
    renderer.render(<Error isOpen={true} message={'test'} />)
    const result = renderer.getRenderOutput()

    expect(result).toMatchSnapshot()
  })
})
