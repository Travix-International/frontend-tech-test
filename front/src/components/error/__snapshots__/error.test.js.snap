// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorComponent Snapshot 1`] = `
<ModalComponent
  onClose={[Function]}
  open={true}
>
  <ForwardRef>
    <ForwardRef>
      Error!
    </ForwardRef>
    <ForwardRef>
      test
    </ForwardRef>
  </ForwardRef>
</ModalComponent>
`;
